pkg/cmd/cli/describe/printer.go

```diff
@@ -138,6 +140,11 @@ func NewHumanReadablePrinter(noHeaders, withNamespace, wide bool, showAll bool,
 	return p
 }
 
+func NewHumanReadablePrinterWithResourceAliases(resourceAliases map[string]string, noHeaders, withNamespace, wide bool, showAll bool, showLabels bool, absoluteTimestamps bool, columnLabels []string) *kctl.HumanReadablePrinter {
+	resourceShortForms = resourceAliases
```

Comment:
> This should be a field of the printer, not mutate the global state (that global variable resourceShortForms above).

There is no straightforward way of doing this without changing the function signature of `NewHumanReadablePrinter` (which would involve changing several other files).

> Calling the function resourceShortFormFor before a call to this NewHumanReadablePrinterWithResourceAliases will range over an empty map, while after the call it will range over whatever was the last thing passed in as argument.

It was the most feasible way I found of doing this, short of re-defining the `shotForms` map in the `shortcut_restmapper.go` file. I cannot import the `clientcmd` package here as it causes a circular dependency.